spec/requests/topics_controller_spec.rb

```diff
@@ -326,6 +326,19 @@
           expect(result['success']).to eq(false)
           expect(result['url']).to be_blank
         end
+
+        it 'returns validation error' do
+          PostCreator.any_instance.stubs(:skip_validations?).returns(false)
+          p1.update_columns(raw: "i", cooked: "")
+          post "/t/#{topic.id}/move-posts.json", params: {
+            post_ids: [p1.id],
+            destination_topic_id: dest_topic.id
+          }
```

Comment:
```suggestion
          PostCreator.any_instance.stubs(:skip_validations?).returns(false)
          p1.update_columns(raw: "i", cooked: "")
          post "/t/#{topic.id}/move-posts.json", params: {
            post_ids: [p1.id],
            title: 'a'
          }
```

I feel like the stub doesn't reflect a real world problem. How about using the `title` attribute instead to trigger the `ActiveRecord::NotSaved` error.

Reply:
It was a good idea. I checked it and this is not going to work. Move to new title is calling create! Which triggers InvalidRecord https://github.com/discourse/discourse/blob/main/app/models/post_mover.rb#L42

When in this scenario, we want to trigger this https://github.com/discourse/discourse/blob/main/lib/post_creator.rb#L257

How this validation was even triggered by discourse-calendar why we have skip_validation=true? The answer is that core is ignoring validations. I guess that assumption is that valid post before is valid post after:

https://github.com/discourse/discourse/blob/main/app/models/post.rb#L65

Plugin is registering new validator which is not checking for skip_validation. Behaviour is inconsistent but also correct, as moving a post with event to an existing topic is not making that post valid anymore

https://github.com/discourse/discourse-calendar/blob/main/plugin.rb#L369

That was the reason for my stub, to somehow fake NotSaved

Question is if  in general, when we move posts, if we should skip validations